cmd/seed-controller-manager/main.go

```diff
@@ -185,6 +185,11 @@ Please install the VerticalPodAutoscaler according to the documentation: https:/
 		log.Fatalw("Unable to create the seed getter", zap.Error(err))
 	}
 
+	seedClientGetter, err := seedClientGetterFactory(rootCtx, mgr.GetClient())
+	if err != nil {
+		log.Fatalw("Unable to create the seed client getter", zap.Error(err))
+	}
+
```

Comment:
I don't think this (and all the other stuff to get this into the EAR controller) is necessary; The seed-controller-manager mainly talks to its assigned Seed, so it only needs one Seed client. You can see other controllers using `mgr.GetClient()` to get a seed client here:

https://github.com/kubermatic/kubermatic/blob/b4a7df6129ef6a978a3f57b926ff24f987f8a429/pkg/controller/seed-controller-manager/preset-controller/controller.go#L69

https://github.com/kubermatic/kubermatic/blob/b4a7df6129ef6a978a3f57b926ff24f987f8a429/pkg/controller/seed-controller-manager/constraint-template-controller/controller.go#L85

https://github.com/kubermatic/kubermatic/blob/b4a7df6129ef6a978a3f57b926ff24f987f8a429/pkg/controller/seed-controller-manager/operating-system-profile-synchronizer/controller.go#L84

Reply:
Using the client embedded into the `Reconciler` struct for the encryption-at-rest-controller should suffice for what the code does, no?

https://github.com/kubermatic/kubermatic/blob/b4a7df6129ef6a978a3f57b926ff24f987f8a429/pkg/controller/seed-controller-manager/encryption-at-rest-controller/controller.go#L90

Reply:
Yes. It should work. I thought seed-ctrl-mgr could run outside the seeds it is managing and then it would be necessary to get the right client. 

But I see no other ctrl hanging on the seed-ctrl-mgr doing this. 

Will cleanup. :+1:

Reply:
Why is there `seedGetter` tho if you already know the seed?

Reply:
I assume because you might want to read information from the `Seed` object (e.g. specific configuration to change controller behaviour).